device/core/src/client.ts

```diff
@@ -483,6 +492,22 @@ export class Client extends EventEmitter {
     this._retryPolicy = policy;
   }
 
+  /*
+   * @method           module:azure-iot-device.Client#setDiagnosticSamplingPercentage
+   * @description      The `setDiagnosticSamplingPercentage` method will set percentage value for diagnostic sampling
+   *
+   * @param {number} value            The value of diagnostic sampling
+   *
+   */
+  setDiagnosticSamplingPercentage(value: number): void {
```

Comment:
>setDiagnosticSamplingPercentage [](start = 2, length = 31)

we already have a setOptions call in the client - why create  a separate API?

Reply:
setOptions set the options in transport level. setDiagnosticSamplingPercentage set the options in diagnosticClient. Since our logic to handle if attach diagnostic info to message is in diagnosticClient, and transport is not related to diagnosticClient. So we can not reuse setOptions here.

Reply:
And the public function to set percentage is just a work around here. Finally we will only update this setting by device twin.